### ISSUE TITLE
Can't update WowUP (app still running even it's closed or uninstalled)

### ISSUE THREAD

**Issue by baronebirra82** [WowUp/WowUp#1175]
**Describe the bug**
For the first time since i installed Windows 11, i had a WoWUp update notification (which prevents me from using the app, so i cannot even start the debug / log functions). If i press the update button i get an errore "WowUp is still running, press OK to close it", if i press OK, after a few seconds it says: "WowUP Cannot Be closed".
I tried everything: closing it in the task manager (but there's nothing there), i uninstalled both via windows and third party uninstaller app, it always says that WowUp is still running.
FYI i got the same exact error while update Warcraft Logs Uploader, which i can't use anymore. Is it a Windows 11 bug or there is a workaround to overcome this? Thanks a lot

**To Reproduce**
1. Update the app

**Expected behavior**
Updating the app.

**Desktop (please complete the following information):**
 - OS: Windows 11

**Comment by linaori** [comment]
As we've seen several of those reports, I have a few questions for you (and everyone else running into this issue):
 - Have you been on the beta in between?
 - Is the update failing from 2.6.2 to 2.7.0?
 - If you uninstall the app and then try to install 2.7.0, does that work?
 - Can you save your logs from `%appdata%\WowUp\logs` and post them here or send them to me or other team members on Discord? DM in case your windows user name is visible, not everyone wants that publicly seen
 - After you've obtained the logs, can you delete `%appdata%\WowUp\` and ensure `%LocalAppData%\Programs\wowup` is gone?
 - Make sure that there's no WowUp process running, if you have WowUp starting up automatically, verify it's not present in startup programs (You can search for "startup apps" in W11) and reboot.

If the above does not fix the issue, if you have time to test, can you see if re-installing [2.6.2](https://github.com/WowUp/WowUp/releases/tag/v2.6.2) works after clearing out all data, and then install the betas to see if any of those cause problems?

You can find the [beta releases here](https://github.com/WowUp/WowUp/tags). To find out if a beta introduced the problem, my advice would be to start with beta9 and work your way down. If you manage to find out if installing a beta works and updating from there seems to go smooth, you can go to Options > Application > (scroll down) > Application Release Channel, and set it back to Stable.

I'm personally using W11 as well, but I've been updating from beta to beta an I've never actually had 2.6.2 installed as the 2.6.2 hotfixes were merged back into the beta.

**Comment by jliddev** [comment]
I also wonder if the standalone version runs without the installer? https://github.com/WowUp/WowUp/releases/download/v2.7.0/WowUp-2.7.0.exe

**Comment by baronebirra82** [comment]
Ok so..first of all, the standalone version works, and updates everything. About the other stuff you've asked me:

Have you been on the beta in between? Never used a beta version
Is the update failing from 2.6.2 to 2.7.0? Yes
If you uninstall the app and then try to install 2.7.0, does that work? Nope, it still says it's running
Can you save your logs from %appdata%\WowUp\logs and post them here or send them to me or other team members on Discord? [DM in case your windows user name is visible, not everyone wants that publicly seen] Logs: https://easyupload.io/jy3ej1
After you've obtained the logs, can you delete %appdata%\WowUp\ and ensure %LocalAppData%\Programs\wowup is gone?
Make sure that there's no WowUp process running, if you have WowUp starting up automatically, verify it's not present in startup programs (You can search for "startup apps" in W11) and reboot.

Anyway it seems that with the standalone everything is working now..still having the same issue with WowAnalyzer. Hope the logs help me figure out what happened there too.

**Comment by jliddev** [comment]
You might give the latest beta  try, it was built with a newer version of Electron Builder.

**Comment by EKirschmann** [comment]
I'm just getting this problem. I can't update or uninstall because this error occurs

![image](https://user-images.githubusercontent.com/35265208/167309183-9b9f30bd-65b2-4aca-9f75-d23e9c33614c.png)

**Comment by Crazyglue** [comment]
Same. Cannot update, via the in-app prompt. I cannot uninstall the app at all (doing so shows the same message, that its running). And downloading the latest version from the site and installing it that way does shows the same message as @EKirschmann 

I've tried looking at the task manager to see if WowUp or anything similar is running, but there are no processes or tasks other than the installer running.

My guess is that the installer is throwing this error cause it thinks the installer is the WowUp application

**Comment by linaori** [comment]
@EKirschmann @Crazyglue is that with the latest beta? Latest being 2.8.0-beta.1: https://github.com/WowUp/WowUp/releases/tag/v2.8.0-beta.1

Additional steps you can try:
 - disable starting wowup when the system starts
 - reboot windows
 - uninstall
 - install

If that doesn't work, try the beta. If that doesn't work, try the portable version (.exe without the setup in the name)

**Comment by EKirschmann** [comment]
Same issue with beta now. can't uninstall 

![image](https://user-images.githubusercontent.com/35265208/168404631-bc176e6c-a3f9-4864-89b2-8f65761fbc95.png)

**Comment by EKirschmann** [comment]
> They never finish?

correct. May or may not be related. If I can figure out how to uninstall or update, maybe I can figure out how to get one of the addons actually updated instead of staying on pending.

**Comment by slowsauce** [comment]
experiencing the exact same problem, tried to update and got this error. i disabled the auto startup and tried to uninstall after restarting and still error.

**Comment by CursedShadow** [comment]
This is still an issue. 

I was updating things on my old laptop for vacation, WoWUP was on 2.5.2. Then this error pops up. No problem, clear the %appdata% folder and try again. Error persists. Whatever. Check WoWup isn't in the start up list. Not there. Restart the laptop, error persists. 

Decide to try and uninstall WoWUP. It seems to go through, no longer shows up on the Windows apps/features list. Cool. I download the WoWUP-Curseforge version from the website and start the install.... Boom, error saying that WoWUp-CF is already running???????????

What? So it let me uninstall but not update or install a whole new installation? Now I'm stuck with no WoWUP and the inability to install WoWUP because somehow the old WoWUP install is holding it hostage from the grave.  What the f*ck?

**Comment by Crazyglue** [comment]
Its been forever, this dumb program has been on my computer for almost a year and I can't get rid of it. So I deleted the folder that it was in and deleted any entries in the registry and I'm calling it a day. Guys, its 2023 now, if you have a program that has an installer, it should have a working uninstaller.

**Comment by jliddev** [comment]
I apologize if my support is not to your liking. There has been an issue with the [Electron Builder](https://www.electron.build/) uninstall process getting stuck due to some file in the install directory for some time. You're welcome to keep track of said issue on your own. https://github.com/electron-userland/electron-builder/issues/6865

**Comment by bucketss** [comment]
still happening in 2024 btw
